### ISSUE TITLE
Add `.deb` publishing

### ISSUE THREAD

**Issue by perlun** [perlang-org/perlang#416]
# Moved to GitLab

Please continue to the new version of this issue here: https://gitlab.perlang.org/perlang/perlang/-/issues/416. The GitHub page you are currently reading will not contain the latest information on this issue.

---

Publishing automated `.tar.gz` files for consumption by [a custom shell script](https://github.com/perlang-org/perlang/blob/master/scripts/perlang-install) is useful, but really, it would be even nicer to have a few `.deb` channels for this:

### `experimental` or `snapshots`

Corresponds to the current content at https://builds.perlang.org/; built automatically for each commit to [the master branch](https://github.com/perlang-org/perlang/commits/master). As the project grows and the number of commits per day potentially make this unfeasible, this could be switched to a nightly release cadence instead. For now, this works perfectly fine.

_Challenge_: How do we generate the Debian changelog for this? Should we create a changelog based on the latest `stable`/`testing` version and just auto-generate a new entry for the `snapshots` version? Or should we somehow fetch the latest snapshot changelog and add an entry to it (this becomes much more complex)?

### `testing`

Corresponds to the currently released versions at https://github.com/perlang-org/perlang/releases. Released every now and then, but much more seldom than the snapshots. This is the version you'd typically use a non-development machine, where you just want something that "works" and typically don't have any grave, major bugs present.

## Where do we host it?

This is a good question. Regretfully, it might be easiest that we just host it on our own infra for now, even though it would be _nice_ to have this on something like GitHub Packages. Unfortunately, it doesn't support support `.deb` packages currently. :neutral_face: We could look into other options also. Just putting it on our own infra _works_, it's just one more service to be responsible for ourselves... so if there was a shared "public transport" for this, it would be really nice to just "take the train" in this case instead of having to "drive your own car". :thinking: Worth considering.

### Webpage instructions

Once this is in place, we should make sure to update the https://perlang.org/download/ page to make mention of the `.deb` package hosting. We'll still keep the `.tar.gz` files for sure; they are useful to non-Debian users. (I consider Ubuntu, Mint and other users to be Debian users here, since they are essentially part of the Debian ecosystem in a broader sense)

**Comment by perlun** [comment]
> **Where do we host it?**
>
> This is a good question. Regretfully, it might be easiest that we just host it on our own infra for now, even though it would be nice to have this on something like GitHub Packages. Unfortunately, it doesn't support support .deb packages currently. 😐

If/when we switch to GitLab, we could hopefully [use its `.deb` repository support](https://docs.gitlab.com/ee/user/packages/debian_repository/index.html), with the caveat that it's [not fully ready](https://gitlab.com/groups/gitlab-org/-/epics/6057) yet.
